src/runtimes/node/detect_es_module.js

```diff
@@ -0,0 +1,22 @@
+const fs = require('fs')
+
+const { init, parse } = require('es-module-lexer')
+
+const detectEsModule = async ({ srcFile }) => {
+  // console.log('srcFile: ', srcFile)
+  if (!srcFile) {
+    return false
+  }
+
+  await init
+
+  // Would love to just use await fs.promises.readFile, but it's not available in our version of Node.
+  // eslint-disable-next-line node/no-sync
+  const srcFileContents = fs.readFileSync(srcFile, { encoding: 'utf8' })
```

Comment:
this could probably use some `try/catch` safety with a default to `false` to safely maintain the status quo